test/can_obd2/can_mapping_test.cpp

```diff
@@ -72,85 +76,128 @@ void CANMappingTest::formula_test(void)
 
 void CANMappingTest::extract_test_bit_mode(void)
 {
-    for (uint8_t length = 1; length <= 32; length++ ) {
-            uint64_t test_value = ((uint64_t)1 << length) - 1;
-            for (uint8_t offset = 0; offset < (CAN_MSG_SIZE * 8) - length + 1; offset++) {
-                    CAN_msg msg;
-                    CANMapping mapping;
-                    memset(&mapping, 0, sizeof(mapping));
-                    memset(&msg, 0, sizeof(CAN_msg));
-                    mapping.offset = offset;
-                    mapping.length = length;
-                    mapping.type = CANMappingType_unsigned;
-                    mapping.bit_mode = true;
-                    msg.data64 = test_value << offset;
-
-                    float value = canmapping_extract_value(msg.data64, &mapping);
+        srand(time(NULL));
+
+        for (size_t random = 0; random <= 1; random++){
```

Comment:
agreed. I was having those feelings of uncertainty about the randomness. Could be good for a fuzzing test, but out of scope here.  I changed it to a bitpattern of 0x55 as an alternative to the all 1's.